app/controllers/free_transactions_controller.rb

```diff
@@ -38,6 +34,16 @@ def create_contact
     contact_form = new_contact_form(params[:listing_conversation])
 
     if contact_form.valid?
+      process_res = TransactionService::API::Api.processes.get(
+        community_id: @current_community.id,
+        process_id: @listing.transaction_type.transaction_process_id
+      )
+
+      unless process_res.data[:process] == "none"
+        flash[:error] = "Sending the message failed. Please try again."
+        return redirect_to root
+      end
```

Comment:
Better style would be to use .maybe and explicitly handle the case of missing transaction_process by raising an error with error message. Now we get only "no method [] on Nil".